### DISCUSSION TITLE
How do I change git's display language?

### DISCUSSION THREAD

**Discussion by User670** [git-for-windows/git#4368] [Q&A] [answered]
Windows 10 x64. My system's displayed language and preferred language are both Chinese, but git displays in English.

Tried googling how to change git language, and every response is based on Linux (add a env variable "LANG=[locale code]"). Somehow I decided to try adding such a variable in my Windows which didn't work.

Is it possible to change Git for Windows's display language in the console? If yes, how?

edit: if it's relevant, I'm using Windows Command Prompt, not Git Bash.

edit: `git version 2.33.1.windows.1`

**Comment by dscho** [comment, marked as answer]
> edit: `git version 2.33.1.windows.1`

You _definitely_ [want to update](https://github.com/git-for-windows/git/wiki/FAQ#should-i-upgrade-to-a-newer-git-for-windows-version).

> Is it possible to change Git for Windows's display language in the console?

This is not supported by Git for Windows out of the box. The translations _are_ built, but they are simply not included in the Git for Windows installer. I zipped them up here: [git-2.40.0-l18n.zip](https://github.com/git-for-windows/git/files/11099746/git-2.40.0-l18n.zip) (watch out, it's ~3MB)

If you unpack them into your installation (you need to do that in an elevated session, e.g. by pressing <kbd>🪟</kbd>, typing "Git Bash", then pressing <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>Enter</kbd>, then invoking `cd / && unzip ~/Downloads/git-2.40.0-l18n.zip`), you can use them by setting the environment variable `LANG`:

```console
$ LANG=zh_CN git log -h
用法：git log [<选项>] [<版本范围>] [[--] <路径>...]
  或：git show [<选项>] <对象>...

    -q, --quiet           不显示差异输出
    --source              显示源
    --use-mailmap         使用邮件映射文件
    --mailmap             --use-mailmap 的别名
    --clear-decorations   清除所有之前定义的修饰过滤器
    --decorate-refs <模式>
                          只修饰与 <模式> 匹配的引用
    --decorate-refs-exclude <模式>
                          不修饰和 <模式> 匹配的引用
    --decorate[=...]      修饰选项
    -L <range:file>       跟踪 <文件> 中 <开始>,<结束> 范围内的行或函数 :<函数名> 的演变
```

**Reply by sixwaaaay** [reply]
Is this file "https://github.com/git-for-windows/git/files/11099746/git-2.40.0-l18n.zip" automatically generated for each release? Where should I find the latest i18n file after each update?

**Reply by dscho** [reply]
@sixwaaaay no, they are not generated automatically. I generated that one manually to determine how much need there is for it, and the first response is 9 months later ;-)

The `.zip` essentially contains the subset of files of the `mingw-w64-git` package that matches the pathspec `mingw64/share/locale/*/LC_MESSAGES/git.mo`.

If you're on the latest Git for Windows version, you could extract the same set of files from https://wingit.blob.core.windows.net/x86-64/mingw-w64-x86_64-git-2.43.0.1.4b968f3ea3-1-any.pkg.tar.xz, or you could simply download the latest version's language file from https://github.com/git-for-windows/git-sdk-64/blob/3bc47ebca0afb5a5eaf1df1cdda0b59419ec30e3/mingw64/share/locale/zh_CN/LC_MESSAGES/git.mo (which assumes that you installed the latest Git for Windows version, otherwise the language file might not match the actual messages of your Git version).

**Comment by goyalyashpal** [comment]
> you need to do that in an elevated session, e.g. by pressing 🪟, typing "Git Bash", then pressing Ctrl+Shift+Enter
> _\- @ dscho at https://github.com/git-for-windows/git/discussions/4368#discussioncomment-5464416_

oh, that's the `super`/os key, it was shown as question mark on my system.
btw, the `<C-S-RET>` to launch elevated right from start menu is neat. thanks for sharing that. 

i see that there's no [`hi`](https://github.com/git-for-windows/git-sdk-64/blob/main/mingw64/share/locale/hi) (`in-hi` or India-Hindi) locale right?

**Reply by dscho** [reply]
Indeed, nobody added that language to https://github.com/git-l10n/git-po/tree/HEAD/po yet. Be sure to consult the `README` in that directory for further information (I don't actually know the details, myself).

**Reply by PhilipOakley** [reply]
> i see that there's no [`hi`](https://github.com/git-for-windows/git-sdk-64/blob/main/mingw64/share/locale/hi) (`in-hi` or India-Hindi) locale right?

Also do have read of the latest [Git Rev News](https://git.github.io/rev_news/2023/11/30/edition-105/) "Developer Spotlight: Alexander Shopov" where they interview the person doing the Bulgarian translation, showing that it's not that hard, just some steady work, to get a translation into Git. -- The fun of Open Source :wink:

**Reply by goyalyashpal** [reply]
awesome. thanks for sharing it. it's definitely motivating. i will look into it.
